lib/trie/trie.go

```diff
@@ -950,6 +985,10 @@ func (t *Trie) clearPrefixAtNode(parent *Node, prefix []byte) (
 // matching the key given in little Endian format.
 // If no node is found at this key, nothing is deleted.
 func (t *Trie) Delete(keyLE []byte) {
+	defer func() {
+		t.handleTrackedDeltas(nil)
+	}()
```

Comment:
```suggestion
	defer t.handleTrackedDeltas(nil)
```

Reply:
Keeping it as is (so I don't forget like in a few hours on another branch) due to https://github.com/ChainSafe/gossamer/pull/2876#discussion_r989956510